cpp/tests/groupby/lists_tests.cu

```diff
@@ -189,8 +188,8 @@ TYPED_TEST(groupby_lists_test, lists_with_nulls)
   auto keys   = lcw<TypeParam> { {{1,1}, {2,2}, {3,3}, {1,1}, {2,2}}, nulls_at({1,2,4})};
   auto values = fwcw<int32_t>  {     0,     1,     2,     3,     4 };
 
-  auto expected_keys   = lcw<TypeParam> { {{1,1}, {null,null}}, null_at(1)};
-  auto expected_values = fwcw<R>        {     3,           7 };
+  auto expected_keys   = lcw<TypeParam> { {{null,null}, {1,1}}, null_at(0)};
```

Comment:
Why did the order change here? Is that an intended side effect?

Reply:
Before this PR, the hash-based groupby uses an algorithm to verify the correctness of results and the order of nulls doesn't matter in that case. The sort-based groupby tests added in this work are using an existing function where nulls have to be placed before non-null elements to make it work. Thus by adding sort-based groupby into lists tests., the reordering of gold reference is expected.

Reply:
My point is that if the ordering is a property of the algorithm (an implementation detail) then we shouldn't write tests that assume any ordering. The tests should sort both the expected values and the actual result and then compare. We should only assume ordering in the test output if the ordering is actually one of the API guarantees.

Reply:
Agree.

Reply:
Filed #12174 to this issue.